src/rhash/hash.c

```diff
@@ -1223,6 +1223,53 @@ static int rc_hash_wii(md5_state_t* md5,
   return 1;
 }
 
+static int rc_hash_wiiware(md5_state_t* md5,
+                           const rc_hash_iterator_t* iterator,
+                           void* file_handle)
+{
+  uint32_t cert_chain_size, ticket_size, tmd_size, data_size;
+  uint32_t tmd_start_addr, hash_size;
+
+  uint8_t quad_buffer[4];
+  uint8_t* buffer;
+
+  /* GetApploaderSize */
+  rc_file_seek(iterator, file_handle, 0x08, SEEK_SET);
+  rc_file_read(iterator, file_handle, quad_buffer, 4);
+  cert_chain_size =
+    ((uint32_t)quad_buffer[0] << 24) | (quad_buffer[1] << 16) | (quad_buffer[2] << 8) | quad_buffer[3];
```

Comment:
The cast to `uint32_t` is redundant (if you don't support `sizeof(int) < 4`), or you need to cast to `uint32_t` before doing `<< 16` (if you do support `sizeof(int) < 4`). Same below.